homeassistant/components/homeconnect/entity.py

```diff
@@ -0,0 +1,71 @@
+"""Home Connect entity base class."""
+
+import logging
+from typing import Optional
+
+from homeassistant.core import callback
+from homeassistant.helpers.dispatcher import async_dispatcher_connect
+from homeassistant.helpers.entity import Entity
+
+from .api import HomeConnectDevice
+from .const import DOMAIN, SIGNAL_UPDATE_ENTITIES
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class HomeConnectEntity(Entity):
+    """Generic Home Connect entity (base class)."""
+
+    def __init__(self, device: HomeConnectDevice, desc: Optional[str]) -> None:
+        """Initialize the entity."""
+        self.device = device
+        self.desc = desc
+        if desc is None:
+            self._name = self.device.appliance.name
+        else:
+            self._name = f"{self.device.appliance.name} {desc}"
+
+    async def async_added_to_hass(self):
+        """Register callbacks."""
+        self.async_on_remove(
+            async_dispatcher_connect(
+                self.hass, SIGNAL_UPDATE_ENTITIES, self._update_callback
+            )
+        )
+
+    @callback
+    def _update_callback(self, ha_id):
+        """Update data."""
+        if ha_id == self.device.appliance.haId:
+            self.async_entity_update()
+
+    @property
+    def should_poll(self):
+        """No polling needed."""
+        return False
+
+    @property
+    def name(self):
+        """Return the name of the node (used for Entity_ID)."""
+        return self._name
+
+    @property
+    def unique_id(self):
+        """Return the unique id base on the id returned by Home Connect and the entity name."""
+        return f"{self.device.appliance.haId}-{self.desc}"
```

Comment:
If `self.desc` is `None` we shouldn't add it. When can it be `None`? Is it just for the power switch? Should we come up with a fake `desc` for that switch to make it future proof if we add more switches in the future?

Reply:
True, that's stupid. I simply use `desc="Power"` now, it's anyway a more appropriate name.